service/src/main/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/UploadAdditionalApplicationsService.java

```diff
@@ -11,40 +11,54 @@
 import uk.gov.hmcts.reform.fpl.model.common.C2DocumentBundle;
 import uk.gov.hmcts.reform.fpl.model.common.Element;
 import uk.gov.hmcts.reform.fpl.model.common.OtherApplicationsBundle;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicListElement;
 import uk.gov.hmcts.reform.fpl.service.time.Time;
 import uk.gov.hmcts.reform.fpl.utils.DocumentUploadHelper;
 
 import java.time.LocalDateTime;
 import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.UUID;
 
 import static java.util.Collections.emptyList;
 import static java.util.Comparator.comparing;
 import static java.util.Comparator.reverseOrder;
 import static org.apache.commons.lang3.ObjectUtils.defaultIfNull;
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
 import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.DATE_TIME;
 import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateTimeBaseUsingFormat;
 
 @Service
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class UploadAdditionalApplicationsService {
 
+    public static final String APPLICANT_SOMEONE_ELSE = "SOMEONE_ELSE";
+
     private final Time time;
     private final DocumentUploadHelper documentUploadHelper;
 
     public AdditionalApplicationsBundle buildAdditionalApplicationsBundle(CaseData caseData) {
+        final Optional<String> applicantName = getSelectedApplicantName(
+            caseData.getApplicantsList(), caseData.getOtherApplicant()
+        );
+
         final String uploadedBy = documentUploadHelper.getUploadedDocumentUserDetails();
         final LocalDateTime currentDateTime = time.now();
 
         C2DocumentBundle c2DocumentBundle = null;
         OtherApplicationsBundle otherApplicationsBundle = null;
 
         if (caseData.getAdditionalApplicationType().contains(AdditionalApplicationType.C2_ORDER)) {
-            c2DocumentBundle = buildC2DocumentBundle(caseData, uploadedBy, currentDateTime);
+            c2DocumentBundle = buildC2DocumentBundle(
+                caseData, applicantName.orElse(EMPTY), uploadedBy, currentDateTime);
         }
 
         if (caseData.getAdditionalApplicationType().contains(AdditionalApplicationType.OTHER_ORDER)) {
-            otherApplicationsBundle = buildOtherApplicationsBundle(caseData, uploadedBy, currentDateTime);
+            otherApplicationsBundle = buildOtherApplicationsBundle(
+                caseData, applicantName.orElse(EMPTY), uploadedBy, currentDateTime);
```

Comment:
When adding the applicant name as an empty string in the case that the getSelectedApplicantName cannot find it does that mean we will have a blank field in the tab?

Reply:
The applicant name field shouldn't be empty as it is a mandatory field. If it is empty, the applicant field will not be displayed on the tab

Reply:
@lewisbirks Updated the code to throw an exception when applicant name is empty